Dockerfile

```diff
@@ -3,48 +3,66 @@ ARG UBUNTU_VERSION=20.04
 FROM ubuntu:${UBUNTU_VERSION} as ubuntu-nodejs
 ARG NODEJS_MAJOR_VERSION=18
 ENV DEBIAN_FRONTEND=nonintercative
-RUN apt-get update && apt-get install curl -y
-RUN curl --proto '=https' --tlsv1.2 -sSf -L https://deb.nodesource.com/setup_${NODEJS_MAJOR_VERSION}.x | bash -
-RUN curl --proto '=https' --tlsv1.2 -sSf -L https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add - &&\
-  echo "deb http://apt.postgresql.org/pub/repos/apt/ `lsb_release -cs`-pgdg main" | tee /etc/apt/sources.list.d/pgdg.list
-RUN apt-get update
-RUN apt-get install nodejs -y
-RUN apt-get install -y --no-install-recommends ca-certificates jq postgresql-client
+
+RUN \
+  apt-get update &&\
+  apt-get install ca-certificates curl gnupg lsb-release -y &&\
+  mkdir -p /etc/apt/keyrings &&\
+  curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg &&\
+  echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODEJS_MAJOR_VERSION.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list &&\
+  curl --proto '=https' --tlsv1.2 -sSf -L https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add - &&\
+  echo "deb http://apt.postgresql.org/pub/repos/apt/ `lsb_release -cs`-pgdg main" | tee /etc/apt/sources.list.d/pgdg.list &&\
+  apt-get update &&\
+  apt-get install nodejs -y &&\
+  apt-get install -y --no-install-recommends ca-certificates jq postgresql-client
 
 FROM ubuntu-nodejs as nodejs-builder
-RUN curl --proto '=https' --tlsv1.2 -sSf -L https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - &&\
+RUN \
+  curl --proto '=https' --tlsv1.2 -sSf -L https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - &&\
   echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list &&\
-  apt-get update && apt-get install pkg-config libusb-1.0 libudev-dev gcc g++ make gnupg2 yarn -y
-RUN yarn global add node-gyp@9.0.0
-RUN mkdir -p /app/packages
+  apt-get update && apt-get install pkg-config libusb-1.0 libudev-dev gcc g++ make gnupg2 yarn -y &&\
+  yarn global add node-gyp@9.0.0
 WORKDIR /app
 COPY build build
-COPY packages packages
+COPY packages/cardano-services/package.json packages/cardano-services/package.json
```

Comment:
Are we interested to copy all packages, or almost all, but some are excluded purposefully?

I'm asking as it may be worth checking if we can replace all those `COPY`s with one line command `COPY packages/*/package.json packages/`

This command will care for every single package, and in return we will have less maintenance of Dockerfile in case of new packages etc.

Reply:
Discussed off-line. This is a bit more complicated than it looks. Agreed to merge this as it is.